torchgen/native_function_generation.py

```diff
@@ -631,11 +631,19 @@ def gen_composite_out_kernel(g: NativeFunctionsGroup) -> Optional[str]:
 
     copy_outs_str = "\n".join(copy_outs)
 
+    if any(
+        str(g.out.func.name).startswith(prefix)
+        for prefix in ("_foreach", "_fused_adam", "_fused_adamw", "_fused_sgd")
+    ):
+        return_stmt_str = f"return {rets[0]}.vec();"
```

Comment:
should this be instead?
I think `unpack` for a TensorList returns a `vector<Tensor>` as per https://github.com/pytorch/pytorch/blob/637cf4a3f2cfdd364005681636ca885bdc4d5887/torch/csrc/autograd/VariableTypeManual.cpp#L106-L116
```suggestion
        return_stmt_str = f"return self_;"
```